### PR TITLE
fuzz: Mock GetRand in process_message(s)

### PR DESCRIPTION
**Pull request by dergoegge**
This PR introduces `g_mock_get_rand` to allow tests to mock `GetRand` and makes `g_mock_get_rand` consume from the fuzzed data in the process_message(s) targets.

Mocking `GetRand` reduces the non-determinism in these targets and results in more stable fuzzing.

### CODE REVIEW

File: src/test/fuzz/process_message.cpp [hunk 1]
```diff
@@ -62,6 +63,11 @@ FUZZ_TARGET(process_message, .init = initialize_process_message)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
 
+    g_mock_get_rand = [&](uint64_t max) -> uint64_t {
+        if (fuzzed_data_provider.remaining_bytes() == 0) return 1;
```
Comment (dergoegge) [inline src/test/fuzz/process_message.cpp:67]: Not having this exception leads to an endless loop here: https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/net_processing.cpp#L5225-L5227

File: src/random.cpp [hunk 2]
```diff
@@ -564,9 +564,11 @@ void RandAddPeriodic() noexcept { ProcRand(nullptr, 0, RNGLevel::PERIODIC); }
 void RandAddEvent(const uint32_t event_info) noexcept { GetRNGState().AddEvent(event_info); }
 
 bool g_mock_deterministic_tests{false};
```
Comment (dergoegge) [inline src/random.cpp:566]: Setting `g_mock_deterministic_tests` is not a great alternative as it makes `GetRand` always return the same value.

**Comment by brunoerg** [comment]
Concept ACK

**Comment by dergoegge** [comment]
> Can you add examples where this matters?

https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/net_processing.cpp#L5369
https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/net_processing.cpp#L1106
https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/net_processing.cpp#L5270
https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/net_processing.cpp#L5264

**Comment by dergoegge** [comment]
> Would be nice to start using one FastRandomContext in PeerManager

This doesn't solve our problem in the fuzz tests since the peerman is global, so this will only help in fuzz tests that create a new peerman each iteration.

**Pull request closed without merging**